### PR TITLE
[New Formula] Heptio Sonobuoy

### PR DESCRIPTION
**Pull request by sc250024**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

 New formula for Heptio Sonobuoy project

### CODE REVIEW

**Comment by sc250024** [comment]
Since this is a new formula, I'm not sure how to add the `bottle do` block. I read the [Bottle documentation](https://github.com/Homebrew/brew/blob/master/docs/Bottles.md), but I'm not sure if I should add the block with empty values, or if the CI process takes care of this.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/sonobuoy.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+                   "./"
+      prefix.install_metafiles
+    end
+  end
+
+  test do
+    output = shell_output("#{bin}/sonobuoy 2>&1")
+    assert_match "Sonobuoy is an introspective kubernetes component that generates reports on cluster conformance", output
+    assert_match version.to_s, shell_output("#{bin}/sonobuoy version 2>&1")
+    output = shell_output("#{bin}/sonobuoy gen 2>&1")
```
[the remark this answers is not in the document]
Comment (sc250024) [reply Formula/sonobuoy.rb]: The `ark` formula has a simple example which just sets some configuration, and then retrieves that configuration values: https://github.com/Homebrew/homebrew-core/blob/master/Formula/ark.rb#L29-L35. 

A test that would function as an example of Sonobuoy's functionality would be to examine a `.tar.gz` that was generated from a test cluster. 

```console
$ sonobuoy e2e --help
Inspect e2e test results. Optionally rerun failed tests

Usage:
  sonobuoy e2e archive.tar.gz [flags]
```

Is it common to commit an entire `.tar.gz` file for test purposes?

[the remark this answers is not in the document]
Comment (fxcoudert) [reply Formula/sonobuoy.rb]: We have some tests that use a `.tar.gz` resource. Look at `adplug` for example.

[the remark this answers is not in the document]
Comment (sc250024) [reply Formula/sonobuoy.rb]: @fxcoudert Turns out the Sonobuoy repo has a `tar.gz` file already used for testing, so we can use that to run `sonobuoy e2e` against.

[the remark this answers is not in the document]
Comment (sc250024) [reply Formula/sonobuoy.rb]: @fxcoudert I've noticed that the majority of formulas use the `stage` method here: https://github.com/Homebrew/brew/blob/master/Library/Homebrew/resource.rb#L68-L76. Is there a method which simply downloads the tarball, but doesn't unpack it? Sonobuoy expects a tarball.

[the remark this answers is not in the document]
Comment (fxcoudert) [reply Formula/sonobuoy.rb]:
```
    r = resource("whatever")
    r.verify_download_integrity(r.fetch)
```

Then use `r.cached_download` as the downloaded path (read-only).

[the remark this answers is not in the document]
Comment (sc250024) [reply Formula/sonobuoy.rb]: Cool, thank you! It's resolved now.

**Comment by sc250024** [comment]
@fxcoudert This should be good to go now, assuming there's nothing else you want tested.

**Comment by sc250024** [comment]
@fxcoudert Is there anything else that's needed for this formula?

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Thanks @sc250024 for the pull request!

**Pull request closed without merging**